server/ingest/http/protocol_test.go

```diff
@@ -70,3 +75,24 @@ func TestRegisterProvider(t *testing.T) {
 		t.Errorf("Error closing indexer core: %s", err)
 	}
 }
+
+func TestAnnounce(t *testing.T) {
+	// Initialize everything
+	ind := test.InitIndex(t, true)
+	reg := test.InitRegistry(t, providerIdent.PeerID)
+	ing := test.InitIngest(t, ind, reg)
+	s := setupServer(ind, ing, reg, t)
+	httpClient := setupClient(s.URL(), t)
+	peerID, privKey, err := providerIdent.Decode()
+	if err != nil {
+		t.Fatal(err)
+	}
+	test.RegisterProviderTest(t, httpClient, peerID, privKey, "/ip4/127.0.0.1/tcp/9999", reg)
+
+	ai, _ := peer.AddrInfoFromString("/ip4/127.0.0.1/tcp/9999/")
+	ai.ID = peerID
+
+	if err := httpClient.Announce(context.Background(), ai, cid.Undef); err != nil {
+		t.Fatal(err)
+	}
+}
```

Comment:
I think this should be moved into `/server/ingest/test/test.go` as `AnnounceTest(...)` and then called from both `/server/ingest/http/protocol_test.go` and `/server/ingest/libp2p/protocol_test.go`